### ISSUE TITLE
nuget version of MQTTnet.Extensions.TopicTemplate is not up to date

### ISSUE THREAD

**Issue by clively** [dotnet/MQTTnet#2055]
### Discussed in https://github.com/dotnet/MQTTnet/discussions/2054

<div type='discussions-op-text'>

<sup>Originally posted by **clively** July 27, 2024</sup>
 MQTTnet.Extensions.TopicTemplate on nuget is not up to date with the latest version.   It is currently showing version 4.3.5.1141.  

As such it is missing things such as the MqttClientSubscribeOptionsBuilder WithTopicTemplate method.   
 
 </div>

The samples don't work because of this.

**Comment by chkr1011** [comment]
I uploaded the correct version. If it does not work, please reopen this ticket.

**Comment by clively** [comment]
When attempting to build the sample project I get the following:

CS1929	'MqttClientSubscribeOptionsBuilder' does not contain a definition for 'WithTopicTemplate' and the best extension method overload 'TopicTemplateExtensions.WithTopicTemplate(MqttTopicFilterBuilder, MqttTopicTemplate, bool)' requires a receiver of type 'MQTTnet.MqttTopicFilterBuilder'	MQTTnet.Samples	C:\<my project details>\Samples\Client\Client_Subscribe_Samples.cs	47	

What I did was pull just that one project from your github repo.  I then changed the project references to target the latest versions of the packages on nuget.   And then I did a build.    I'm attaching a screen shot.

It looks like the MqttClientSubscribeOptionsBuilder WithTopicTemplate extension method was not compiled into the package.

![mqttnetissues](https://github.com/user-attachments/assets/adcbcead-97db-4605-a8e8-449fb658e8b0)
